### PR TITLE
Fix destroy testing.

### PR DESCRIPTION
**Pull request by paddycarver**
We weren't properly accounting for TestStep.Destroy in our new
acceptance testing driver. This commit restores the behavior of v1 with
regards to destroy testing.

It also restores other minor areas where we drifted from the previous
testing behaviors.

It fixes an undiscovered bug where TestStep.ExpectNonEmptyPlan wouldn't
fail if an empty plan was returned.

### CODE REVIEW

**Comment by paddycarver** [comment]
I believe this should supersede #544.

**Comment by paddycarver** [comment]
This relies on hashicorp/terraform-plugin-test#37, as well.

**Review by kmoe — APPROVED** [review]
hashicorp/terraform-plugin-test#37 is released in terraform-plugin-test v2.1.0.

LGTM pending manual test approval from @megan07.

**Pull request merged**
